Dockerfile-ngc-pytorch

```diff
@@ -0,0 +1,53 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+RUN mkdir -p /var/run/sshd
+
+# Copy various shell scripts that group dependencies for install
+# Then, make sure that these scripts have execute permissions
+COPY dockerfile_scripts /tmp/det_dockerfile_scripts
+RUN chmod -R 777 /tmp
+
+# Install deb packages
+RUN /tmp/det_dockerfile_scripts/install_deb_packages.sh
+
+# We uninstall these packages after installing. This ensures that we can
+# successfully install these packages into containers running as non-root.
+# `pip` does not uninstall dependencies, so we still have all the dependencies
+# installed.
+RUN python -m pip install determined && python -m pip uninstall -y determined
+
+# Install additional pytorch requriements
+RUN python -m pip install -r /tmp/det_dockerfile_scripts/additional-requirements-torch.txt
+
+# Install jupyter notebook requirements
+RUN python -m pip install -r /tmp/det_dockerfile_scripts/notebook-requirements.txt
+ENV JUPYTER_CONFIG_DIR=/run/determined/jupyter/config
+ENV JUPYTER_DATA_DIR=/run/determined/jupyter/data
+ENV JUPYTER_RUNTIME_DIR=/run/determined/jupyter/runtime
+
+RUN /tmp/det_dockerfile_scripts/add_det_nobody_user.sh
+
+CMD ["/bin/bash"]
+
+ARG AWS_PLUGIN_INSTALL_DIR=/container/aws
```

Comment:
I think the AWS plugin bits can go into the -hpc- version of the targets that PAD will add since it's specific to ss11.

Reply:
We need this for our other image offerings. (i.e. Pytorch 2.0.1 and Pytorch 1.12) But this can be removed once we make these NGC images the default.

Reply:
Does PyTorch need the AWS plugin bits though? AFAIK they’re added by us for the HPC environments where the interconnect is not IB. Seems we could leave that part in the HPC env.